include/nuttx/net/net.h

```diff
@@ -155,7 +155,8 @@ enum net_lltype_e
   NET_LL_BLUETOOTH,    /* Bluetooth */
   NET_LL_IEEE80211,    /* IEEE 802.11 */
   NET_LL_IEEE802154,   /* IEEE 802.15.4 MAC */
-  NET_LL_PKTRADIO      /* Non-standard packet radio */
+  NET_LL_PKTRADIO,     /* Non-standard packet radio */
+  NET_LL_MBIM
```

Comment:
I don't think this is correct.

Reply:
What concerns me most is that NET_LL_MBIN is defined but used only in netdev_register().  For other MAC layers, there are numerous references throughout the networking (see below).  My gut says that the the NET_LL_MBIM must be integrated into the network in many of these places too.  Certainly ALL of these must be reviewed and those that require modification must be fixed:

    $ grep -rl NET_LL_ net
    net/arp/arp_send.c
    net/arp/arp_table.c
    net/bluetooth/bluetooth_finddev.c
    net/bluetooth/bluetooth_recvfrom.c
    net/bluetooth/bluetooth_sendto.c
    net/devif/devif_forward.c
    net/devif/devif_poll.c
    net/devif/ipv4_input.c
    net/devif/ipv6_input.c
    net/icmp/icmp_input.c
    net/icmp/icmp_poll.c
    net/icmp/icmp_recvfrom.c
    net/icmp/icmp_sendto.c
    net/icmpv6/icmpv6_advertise.c
    net/icmpv6/icmpv6_input.c
    net/icmpv6/icmpv6_poll.c
    net/icmpv6/icmpv6_radvertise.c
    net/icmpv6/icmpv6_recvfrom.c
    net/icmpv6/icmpv6_rsolicit.c
    net/icmpv6/icmpv6_sendto.c
    net/icmpv6/icmpv6_solicit.c
    net/ieee802154/ieee802154_finddev.c
    net/ieee802154/ieee802154_recvfrom.c
    net/ieee802154/ieee802154_sendto.c
    net/igmp/igmp_input.c
    net/igmp/igmp_poll.c
    net/igmp/igmp_send.c
    net/ipforward/ipfwd_forward.c
    net/ipforward/ipfwd_poll.c
    net/ipforward/ipv4_forward.c
    net/ipforward/ipv6_forward.c
    net/mld/mld_poll.c
    net/mld/mld_query.c
    net/mld/mld_send.c
    net/neighbor/neighbor_ethernet_out.c
    net/neighbor/neighbor_out.c
    net/netdev/netdev_default.c
    net/netdev/netdev_ioctl.c
    net/netdev/netdev_lladdrsize.c
    net/netdev/netdev_register.c
    net/pkt/pkt_poll.c
    net/pkt/pkt_sockif.c
    net/procfs/netdev_statistics.c
    net/sixlowpan/sixlowpan_framelist.c
    net/sixlowpan/sixlowpan_send.c
    net/sixlowpan/sixlowpan_tcpsend.c
    net/sixlowpan/sixlowpan_udpsend.c
    net/sixlowpan/sixlowpan_utils.c
    net/tcp/tcp_conn.c
    net/tcp/tcp_input.c
    net/tcp/tcp_ipselect.c
    net/tcp/tcp_recvfrom.c
    net/tcp/tcp_recvwindow.c
    net/tcp/tcp_send.c
    net/tcp/tcp_sendfile.c
    net/tcp/tcp_send_buffered.c
    net/tcp/tcp_send_unbuffered.c
    net/udp/udp_callback.c
    net/udp/udp_conn.c
    net/udp/udp_input.c
    net/udp/udp_ipselect.c
    net/udp/udp_recvfrom.c
    net/udp/udp_send.c
    net/udp/udp_sendto_buffered.c
    net/utils/net_icmpchksum.c
    net/utils/net_ipchksum.c

Reply:
Certainly some of those hits are not relevant (liked sixlowpan).  But others point out significant missing logic that must be fixed.

Reply:
The integration into net/procfs/netdev_statistics.c is essential or else the ifconfig command will not work.  net/netdev/netdev_lladdrsize.c is used to report the size of the MAC header.

I don't think any of the forwarding applies.  That is only when we have to reformat packets in order to forward them.  I would need to check, but I think that the IOCTL stuff is only necessary for MAC-specific IOCTL support.

I would have to examine the remaining to see what is and is not relevant.

Reply:
Related changes are also required in apps/ for selecting the right protocols and device names.  Perhaps other things.  apps/netutils/netlib will need to be checked.